pkg/bbgo/session.go

```diff
@@ -81,22 +81,37 @@ func NewStandardIndicatorSet(symbol string, store *MarketDataStore) *StandardInd
 		// setup boll indicator, we may refactor boll indicator by subscribing SMA indicator,
 		// however, since general used BOLLINGER band use window 21, which is not in the existing SMA indicator sets.
 		// Pull out the bandwidth configuration as the boll Key
-		iw := types.IntervalWindow{Interval: interval, Window: 21}
-		set.boll[iw] = &indicator.BOLL{IntervalWindow: iw, K: 2.0}
-		set.boll[iw].Bind(store)
+		iwTmp := types.IntervalWindow{Interval: interval, Window: 21}
+		iwb := types.IntervalWindowBandWidth{IntervalWindow: iwTmp, BandWidth: 1.0}
+		set.boll[iwb] = &indicator.BOLL{IntervalWindow: iwTmp, K: 1.0}
+		set.boll[iwb].Bind(store)
 	}
 
 	return set
 }
 
-// BOLL returns the bollinger band indicator of the given interval and the window,
-// Please note that the K for std dev is fixed and defaults to 2.0
+// BOLL returns the bollinger band indicator of the given interval, the window and bandwidth
 func (set *StandardIndicatorSet) BOLL(iw types.IntervalWindow, bandWidth float64) *indicator.BOLL {
-	inc, ok := set.boll[iw]
+
+	iwb := types.IntervalWindowBandWidth{IntervalWindow: iw, BandWidth: bandWidth}
+	inc, ok := set.boll[iwb]
+
 	if !ok {
 		inc = &indicator.BOLL{IntervalWindow: iw, K: bandWidth}
+
+		// Bind tmp data store
+		tmpDataStore := NewMarketDataStore("TMP")
+		inc.Bind(tmpDataStore)
+
+		// Trigger tmp data store to calculate indicator
+		Klines := set.store.KLineWindows[iw.Interval]
```

Comment:
you can see how did we handle EWMA, I think they are similiar.

Reply:
@c9s But EWMA only support `7, 25, 99`, original bollinger band only support 20, 2

I just want bollinger support dynamic interval and band width at run time.

Currently if I want to setup bollinger bands like 12, 1 at [bollmaker strategy](https://github.com/c9s/bbgo/blob/main/pkg/strategy/bollmaker/strategy.go#L511-L512), I need to change code at `NewStandardIndicatorSet` method and compile again.

Because The bollinger is initialized at [NewStandardIndicatorSet](https://github.com/c9s/bbgo/blob/main/pkg/bbgo/session.go#L47) when session setup, and then update kline to calculate the bollinger value. 

When run bollmaker  strategy, I found the session was already initialized, so the bollinger 12, 1 creator [BOLL](https://github.com/c9s/bbgo/blob/main/pkg/bbgo/session.go#L95) cannot calculate value with the old kline data. The dynamic bollinger  12, 1 that do not match 20, 2 have zero value at beginning.